### PR TITLE
fix(wallet-api): add domain validation for customDappUrl [LIVE-24770] 

### PR DESCRIPTION
**Pull request by Justkant**
### ✅ Checklist

- [x] `npx changeset` was attached.
- [x] **Covered by automatic tests.**
- [x] **Impact of the changes:**
  - deeplink customDappUrl

### 📝 Description

Add security checks to ensure customDappUrl can only be applied when it matches the domain of the original manifest URL (params.dappUrl or manifest.url for dapp manifests).

Changes:
- Create shared manifestDomainUtils with getDomain, isSameDomain, and applyCustomDappUrl functions
- Create useLiveAppManifest hook to centralize manifest fetching and domain validation logic
- Update desktop LiveApp to use shared hook with domain validation
- Update mobile LiveApp to use shared hook with domain validation
- Add comprehensive test suite (33 tests) covering all utility functions and edge cases

This prevents potential security issues from cross-domain URL manipulation while maintaining backward compatibility for same-domain URL customization.

### ❓ Context

- **JIRA or GitHub link**: [LIVE-24770] 

---

### 🧐 Checklist for the PR Reviewers

- **The code aligns with the requirements** described in the linked JIRA or GitHub issue.
- **The PR description clearly documents the changes** made and explains any technical trade-offs or design decisions.
- **There are no undocumented trade-offs**, technical debt, or maintainability issues.
- **The PR has been tested** thoroughly, and any potential edge cases have been considered and handled.
- **Any new dependencies** have been justified and documented.
- **Performance** considerations have been taken into account. (changes have been profiled or benchmarked if necessary)

[LIVE-24770]: https://ledgerhq.atlassian.net/browse/LIVE-24770?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: libs/ledger-live-common/src/wallet-api/manifestDomainUtils.test.ts [hunk 1]
```diff
@@ -0,0 +1,262 @@
...
+      expect(isSameDomain(undefined, undefined)).toBe(false);
+    });
+
+    it("should return false when first URL is invalid", () => {
+      expect(isSameDomain("not-a-url", "https://example.com/path")).toBe(false);
+    });
+
+    it("should return false when second URL is invalid", () => {
+      expect(isSameDomain("https://example.com/path", "not-a-url")).toBe(false);
+    });
```
Comment (francois-guerin-ledger) [inline libs/ledger-live-common/src/wallet-api/manifestDomainUtils.test.ts:135]: Can we test equal strings that are invalid addresses ?

**Review by francois-guerin-ledger — DISMISSED** [review]

File: libs/ledger-live-common/src/wallet-api/manifestDomainUtils.ts [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+): T | null | undefined {
+  if (!customDappUrl || !manifest) {
+    return manifest;
+  }
+
+  // Handle manifest with params.dappUrl
+  if (
+    manifest.params &&
+    "dappUrl" in manifest.params &&
+    isSameDomain(customDappUrl, manifest.params.dappUrl)
```
Comment (b0l0k) [inline libs/ledger-live-common/src/wallet-api/manifestDomainUtils.ts:61]: We should add a comparison on the protocol/port as well (https vs http vs wss).

Comment (b0l0k) [reply libs/ledger-live-common/src/wallet-api/manifestDomainUtils.ts:61]: Actually, it really depends on the use case, so please, consider it if it is relevant.

File: libs/ledger-live-common/src/wallet-api/useLiveAppManifest.ts [hunk 3]
```diff
@@ -0,0 +1,16 @@
+import { useMemo } from "react";
+import { useLocalLiveAppManifest } from "./LocalLiveAppProvider/index";
+import { useRemoteLiveAppManifest } from "../platform/providers/RemoteLiveAppProvider/index";
+import { applyCustomDappUrl } from "./manifestDomainUtils";
+
+export function useLiveAppManifest(appId: string | undefined, customDappUrl?: string | null) {
+  const localManifest = useLocalLiveAppManifest(appId);
+  const remoteManifest = useRemoteLiveAppManifest(appId);
+
+  const manifest = useMemo(() => {
```
Comment (b0l0k) [inline libs/ledger-live-common/src/wallet-api/useLiveAppManifest.ts:10]: Do we really need a memo here?

File: libs/ledger-live-common/src/wallet-api/manifestDomainUtils.test.ts (hunk 1 above)
Comment (Justkant) [reply libs/ledger-live-common/src/wallet-api/manifestDomainUtils.test.ts:135]: You mean adding a test to cover both url invalid ?

File: libs/ledger-live-common/src/wallet-api/manifestDomainUtils.ts (hunk 2 above)
Comment (Justkant) [reply libs/ledger-live-common/src/wallet-api/manifestDomainUtils.ts:61]: Yes I'll check that

File: libs/ledger-live-common/src/wallet-api/useLiveAppManifest.ts (hunk 3 above)
Comment (Justkant) [reply libs/ledger-live-common/src/wallet-api/useLiveAppManifest.ts:10]: It will avoid some of the other hooks down the line depending on the manifest to trigger again on every render, not a big impact here and I also prefer deriving when possible instead of memoizing for no reason
